hazelcast-jclouds/src/main/java/com/hazelcast/jclouds/ComputeServiceBuilder.java

```diff
@@ -144,7 +146,6 @@ ComputeService build() {
         final String identity = getOrNull(JCloudsProperties.IDENTITY);
         String credential = getOrNull(JCloudsProperties.CREDENTIAL);
         final String credentialPath = getOrNull(JCloudsProperties.CREDENTIAL_PATH);
-
         checkNotNull(cloudProvider, "Cloud provider is not set");
         checkNotNull(identity, "Cloud provider identity is not set");
```

Comment:
Yes, right moved checks here : https://github.com/hazelcast/hazelcast/pull/6476/files#diff-627080c6781ee2f36b9958aafc799128R256 If iam-role is set no identity or credential should be set. If aim-role is not set identity or credentials must be present.